devimint/src/federation.rs

```diff
@@ -219,20 +219,17 @@ impl Federation {
 
     pub async fn await_block_sync(&self) -> Result<u64> {
         let finality_delay = self.get_finality_delay().await?;
-        let bitcoind_block_count = self.bitcoind.client().get_blockchain_info()?.blocks;
-        let expected = bitcoind_block_count.saturating_sub(finality_delay.into());
+        // Fedimint's IBitcoindRpc considers block count the total number of blocks,
+        // where bitcoind's rpc returns the height. Since the genesis block has height
+        // 0, we need to add 1 to get the total block count.
+        let block_count = self.bitcoind.client().get_block_count()? + 1;
```

Comment:
**suggestion:** Could we just do `get_block_count` on `Bitcoind` itself and put the adjustment there?

Reply:
Ahh yep, that's a good improvement :+1: 

https://github.com/fedimint/fedimint/pull/3898